include/deal.II/lac/scalapack.h

```diff
@@ -0,0 +1,516 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2017 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_scalapack_h
+#define dealii_scalapack_h
+
+#include <deal.II/base/config.h>
+
+#ifdef DEAL_II_WITH_SCALAPACK
+
+#include <deal.II/base/exceptions.h>
+#include <deal.II/lac/full_matrix.h>
+#include <deal.II/lac/lapack_support.h>
+#include <deal.II/base/mpi.h>
+#include <mpi.h>
+
+#include <memory>
+
+DEAL_II_NAMESPACE_OPEN
+
+
+// Forward declaration of class ScaLAPACKMatrix for ProcessGrid
+template <typename NumberType> class ScaLAPACKMatrix;
+
+
+/**
+ * A class taking care of setting up the two dimensional process grid.
+ * For example an MPI communicator with 5 cores can be arraged into a
```

Comment:
MPI really only talks about processes, not cores, so I stumbled here. Maybe say "5 MPI processes"?